### DISCUSSION TITLE
x-forwarded-for gets overwritten when deployed in Vercel

### DISCUSSION THREAD

**Discussion by miguelvictor** [vercel/community#2484] [Help] [answered]
Hello!

We're calling third-party APIs from our Next.js app deployed in Vercel and we noticed that the value of some headers, especially the `x-forwarded-for` header is being replaced when deployed in Vercel. When we run the app locally, this doesn't happen.

**Comment by Pieparker** [comment, marked as answer]
Hey! Restrictions are in place for this header, to prevent IP spoofing. For customers on the Enterprise plan, a feature is available for Vercel to trust your X-Forwarded-For IP. If this may be of interest to you, [contact sales](https://vercel.com/contact/sales).

([Source](https://vercel.com/docs/concepts/edge-network/headers#x-forwarded-for))
